docs/src/pages/components/text/TextPropControls.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+
+import {
+  Flex,
+  SelectField,
+  SliderField,
+  SwitchField,
+  TextField,
+  TextProps,
+  TextVariation,
+} from '@aws-amplify/ui-react';
+
+export interface TextPropControlsProps extends TextProps {
+  setAs: (value: React.SetStateAction<TextProps['as']>) => void;
+  setVariation: (value: React.SetStateAction<TextProps['variation']>) => void;
+  setIsTruncated: (
+    value: React.SetStateAction<TextProps['isTruncated']>
+  ) => void;
+  setColor: (value: React.SetStateAction<TextProps['color']>) => void;
+  setLineHeight: (value: React.SetStateAction<TextProps['lineHeight']>) => void;
+  setFontWeight: (value: React.SetStateAction<TextProps['fontWeight']>) => void;
+  setFontStyle: (value: React.SetStateAction<TextProps['fontStyle']>) => void;
+  setFontSize: (value: React.SetStateAction<TextProps['fontSize']>) => void;
+  setTextDecoration: (
+    value: React.SetStateAction<TextProps['textDecoration']>
+  ) => void;
+  value: string;
+  setValue: (value: string) => void;
+}
+
+interface TextPropControlsInterface {
+  (props: TextPropControlsProps): JSX.Element;
+}
+
+export const TextPropControls: TextPropControlsInterface = ({
+  variation,
+  setVariation,
+  as,
+  setAs,
+  isTruncated,
+  setIsTruncated,
+  color,
+  setColor,
+  lineHeight,
+  setLineHeight,
+  fontWeight,
+  setFontWeight,
+  fontSize,
+  setFontSize,
+  fontStyle,
+  setFontStyle,
+  textDecoration,
+  setTextDecoration,
+  value,
+  setValue,
+}) => {
+  const VARIATIONS_OPTIONS = [
+    'primary',
+    'secondary',
+    'tertiary',
+    'error',
+    'warning',
+    'info',
+    'success',
+  ];
+
+  const AS_OPTIONS = ['p', 'span', 'strong', 'em'];
+
+  return (
+    <Flex direction="column">
```

Comment:
Can we make these prop fields in the same order as the props to code?

![Screen Shot 2022-04-19 at 4 38 30 PM](https://user-images.githubusercontent.com/48109584/164106200-8b8e602b-d6f7-4c87-8f24-43096485c3fd.png)

Reply:
Done. I re-ordered the code demo instead.